src/main/scala/ch/epfl/bluebrain/nexus/kg/async/DistributedCache.scala

```diff
@@ -183,61 +175,74 @@ trait DistributedCache[F[_]] {
     *
     * @param ref       the project reference
     * @param view      the view to add
-    * @param instant   the instant used to merge the register value
     * @param updateRev whether to update the view collection if the view provided has a higher revision than an
     *                  already existing element in the collection with the same id
     * @return true if the update was performed or false if the element was already in the set
     */
-  def addView(ref: ProjectRef, view: View, instant: Instant, updateRev: Boolean): F[Boolean]
+  def addView(ref: ProjectRef, view: View, updateRev: Boolean): F[Boolean]
 
   /**
     * Removes the view identified by the argument id from the collection of project views.
     *
-    * @param ref     the project reference
-    * @param id      the id of the view to remove
-    * @param instant the instant used to merge the register value
+    * @param ref the project reference
+    * @param id  the id of the view to remove
+    * @param rev revision of the deprecated view
     * @return true of the removal was performed or false of the element was not in the set
     */
-  def removeView(ref: ProjectRef, id: AbsoluteIri, instant: Instant): F[Boolean]
+  def removeView(ref: ProjectRef, id: AbsoluteIri, rev: Long): F[Boolean]
 
   /**
     * Either adds, updates or removes the argument view depending on its deprecation state, revision and the current
     * state of the register.
     *
     * @param ref     the project reference
     * @param view    the view
-    * @param instant the instant used to merge the register value
     * @return true if an update has taken place, false otherwise
     */
-  def applyView(ref: ProjectRef, view: View, instant: Instant): F[Boolean] =
-    if (view.deprecated) removeView(ref, view.id, instant)
-    else addView(ref, view, instant, updateRev = true)
+  def applyView(ref: ProjectRef, view: View): F[Boolean] =
+    if (view.deprecated) removeView(ref, view.id, view.rev)
+    else addView(ref, view, updateRev = true)
 }
 
 object DistributedCache {
 
-  private[async] def accountKey(ref: AccountRef): LWWRegisterKey[RevisionedValue[Option[Account]]] =
+  private[async] sealed trait RevisionedId[A] {
```

Comment:
I would suggest implementing this as a type class (see https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/Show.scala) and provide instances in the respective companion objects.

Reply:
You can also move it to the package level as we can expect uses outside the scope of the cache.